Toolbox image build is failing

I also suspect a chicken-and-egg problem is preventing this from resolving naturally.

The error:
```
 Total                                           5.7 MB/s |  34 MB     00:05     
Running transaction check
Transaction check succeeded.
Running transaction test
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Transaction test error:
  file /usr/bin/systemd-tmpfiles from install of systemd-250.3-8.fc36.x86_64 conflicts with file from package systemd-standalone-tmpfiles-250.9-1.fc36.x86_64

The command '/bin/sh -c dnf -y install samba-test' returned a non-zero code: 1
make: *** [Makefile:138: .build.toolbox] Error 1
```

I was unable to reproduce this locally which leads me to thinking there's a chicken and egg problem. What I suspect is that at some point the fedora base image and the systemd update conflicted in the image that was created as samba-server:latest. Now every time it builds it builds based on the samba-server:latest on quay.io which has the problem. Then the build fails when the build fails a newer samba-server:latest is _not_ pused to quay because all jobs must succeed to trigger the push.

I see two options:
1) We do not require all jobs to run for push. Each "chain" of actions relates to one image
2) During the build the job(s) responsible for building the toolbox share/reuse the samba-server image that was just built. This may mean combining the steps responsible for building those images.

I prefer item 2 above. It may seem a messy on first glance, but it reflects an underlying relationship between those images.

--- a/.github/workflows/container-image.yml
+++ b/.github/workflows/container-image.yml
@@ -70,13 +70,27 @@ jobs:
     - uses: actions/checkout@v3
     - name: Build the client image
       run: make build-client
+    # Here we upload samba-client image to artifacts locally for consumption
+    # during samba-toolbox build process.
+    - name: Upload client image
+      uses: ishworkh/docker-image-artifact-upload@v1
+      with:
+        image: "quay.io/samba.org/samba-client:latest"
+        retention_days: 1
 
   build-toolbox:
+    needs: build-client
     runs-on: ubuntu-latest
     env:
       BUILDAH_FORMAT: oci
     steps:
     - uses: actions/checkout@v3
+    # Download locally stored samba-client image to be used as base for building
+    # samba-toolbox.
+    - name: Download client image
+      uses: ishworkh/docker-image-artifact-download@v1
+      with:
+        image: "quay.io/samba.org/samba-client:latest"
     - name: Build the toolbox image
       run: make build-toolbox
 

--- a/images/client/Containerfile
+++ b/images/client/Containerfile
@@ -4,4 +4,11 @@ FROM registry.fedoraproject.org/fedora:36
 
 MAINTAINER Michael Adam <obnox@samba.org>
 
-RUN dnf -y install samba-client
+# https://github.com/samba-in-kubernetes/samba-container/issues/96#issuecomment-1387467396
+#
+# samba-common, when pulled in as a dependency for samba-client, has a preferred
+# requirement on systemd-standalone-tmpfiles(rather than systemd) but is meant
+# to conflict with systemd itself of same version or higher. We can avoid the
+# conflict by choosing to install systemd over systemd-standalone-tmpfiles.
+RUN dnf -y install samba-client systemd \
+    && dnf clean all

--- a/images/toolbox/Containerfile
+++ b/images/toolbox/Containerfile
@@ -1,3 +1,4 @@
-FROM registry.fedoraproject.org/fedora:36
+FROM quay.io/samba.org/samba-client:latest
 MAINTAINER Shachar Sharon <ssharon@redhat.com>
-RUN dnf -y install samba-client samba-test
+RUN dnf -y install samba-test \
+    && dnf clean all

